### PR TITLE
Update 0.21.3

### PR DESCRIPTION
**Pull request by SC426**
[Upstream](https://github.com/jelmer/dulwich)
[Changelog](https://github.com/jelmer/dulwich/releases)

### Actions
- Update version number and sha256
- Add `py<38` build skip
- Update dependencies
- Linter fixes

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,26 +24,17 @@ requirements:
   host:
     - python
     - pip
-    - urllib3
-    # secure
-    - pyopenssl >=0.14
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:28]: on CF all those deps are still there, are they really not needed ?

Comment (SC426) [reply recipe/meta.yaml:28]: I don't see anything else listed in the upstream repo

https://github.com/jelmer/dulwich/blob/2c9cf9de8266aee2b966fb1f89eb6c55fbcc7a23/pyproject.toml#L2

https://github.com/jelmer/dulwich/blob/master/requirements.txt

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:28]: see this file: https://github.com/conda-forge/dulwich-feedstock/blob/main/recipe/meta.yaml

File: recipe/meta.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ source:
 build:
   number: 0
   script: {{ PYTHON }} -m pip install . --no-deps -vv
+  skip: true  # [py<38]
 
 requirements:
   build:
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:20]: Remove lines:
```
    - python                                 # [build_platform != target_platform]
    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -23,26 +24,17 @@ requirements:
...
     - python
-    - urllib3
-    # secure
-    - pyopenssl >=0.14
-    - cryptography >=1.3.4
-    - idna >=2.0.0
-    - certifi
-    - ipaddress  # [py<=33]
-    - pysocks >=1.5.6,<2.0,!=1.5.7
+    - urllib3 >=1.23
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Should be `urllib3 >=1.25` https://github.com/jelmer/dulwich/blob/dulwich-0.21.3/setup.cfg#LL54C5-L54C18

File: recipe/meta.yaml [hunk 4]
```diff
@@ -23,26 +24,17 @@ requirements:
...
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:32]: It makes sense to add:
```
  run_constrained:
    - paramiko
```
see https://github.com/jelmer/dulwich/blob/d78de71a390e349106e7fb78e87255706dc75db9/setup.cfg#LL38C12-L38C20

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:28]: Those dependencies are from 2018 https://github.com/conda-forge/dulwich-feedstock/commit/b4afda2e5c3827d2eaed4631524e9ba5fcefd5cf and now they are not required

File: recipe/meta.yaml (hunk 2 above)
Comment (SC426) [reply recipe/meta.yaml:20]: Done! Thanks

File: recipe/meta.yaml (hunk 3 above)
Comment (SC426) [reply recipe/meta.yaml]: Fixed

File: recipe/meta.yaml (hunk 4 above)
Comment (SC426) [reply recipe/meta.yaml:32]: Added!

**Review by ELundby45 — APPROVED** [review]

**Pull request merged**